### PR TITLE
FOUR-18130: [FALL] Case Detail Page IT4

### PR DESCRIPTION
**Pull request by pmPaulis**
## Issue & Reproduction Steps
The updated Case Detail Page addresses the core pain points of fragmented information, improve tracking, and visibility. By introducing a unified and comprehensive view, it enhances efficiency, organization, tracking, and productivity in Case management. (In Flight Process Map, Case Timeline, Cancel Behavior, Counters)

**PRD**
[Case Detail Page](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/3515285505/Case+Detail+Page) 

**FIGMA**
[Fall 2024](https://www.figma.com/design/Jfl3HAOx28TVHMl7cHJNIP/Fall-2024?node-id=376-3235&t=T9PI32sKctk90jRc-0)

## How to Test
Go to **Cases** menu

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-18130

## PR Related
https://github.com/ProcessMaker/connector-send-email/pull/325
https://github.com/ProcessMaker/package-webentry/pull/253
https://github.com/ProcessMaker/package-comments/pull/132

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:connector-send-email:feature/FOUR-19490
ci:package-webentry:feature/FOUR-19490
ci:package-comments:feature/FOUR-18130-fall

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-c652896a1a.engk8s.processmaker.net

File: resources/jscomposition/cases/casesDetail/components/EllipsisMenu.vue [hunk 1]
```diff
@@ -1,54 +1,49 @@
...
-    <div class="tw-float-right tw-relative tw-right-4">
-      <button
-        v-for="item in menuOptions"
-        :key="item.menu"
-        type="button"
-        class="tw-rounded-lg tw-bg-white tw-px-2 tw-py-1 tw-text-xs tw-font-semibold tw-text-gray-600 tw-shadow-sm
+  <div class="tw-float-right tw-absolute tw-right-4 tw-top-0">
+    <button
+      v-for="item in menuOptions"
+      v-if="row.screenId"
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/EllipsisMenu.vue:5]: this must be in different tags
https://vuejs.org/style-guide/rules-essential.html#avoid-v-if-with-v-for

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        initials: "S",
+      },
+    };
+  },
+  computed: {
+    disabled() {
+      return this.form.subject.trim() === "" || this.form.body.trim() === "";
+    },
+  },
+  watch: {},
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:86]: remove empty

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      const that = this;
+      that.form.commentable_id = that.commentable_id;
+      that.form.commentable_type = that.commentable_type;
+      that.form.type = that.type ? that.type : "MESSAGE";
+      that.form.hidden = that.hidden ? that.hidden : false;
+      ProcessMaker.apiClient.post("comments", that.form).then((response) => {
+        that.load();
+        that.emptyForm();
+      });
+    },
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:132]: This method is not using

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+</script>
+
+<style>
+#systemAvatar {
+  max-height: 24px;
+  max-width: 24px;
+  border-radius: 50%;
+  margin-left: 3px;
+}
+
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:144]: remove blank line

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+.row .col-auto {
+  height: 3em;
+}
+
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:148]: remove blank line

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+.date {
+  color: #848484;
+  font-size: 0.8em;
+}
+
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:153]: remove blank line

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+.comment-body {
+  line-height: 1.2em;
+}
+
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:157]: remove blank line

File: resources/jscomposition/cases/casesDetail/components/TimelineCases.vue [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    to right,
+    transparent 0%,
+    transparent 38px,
+    rgb(230, 230, 230) 38px,
+    rgb(230, 230, 230) 40px,
+    transparent 40px,
+    transparent 100%
+  );
+}
+
```
Comment (fagubla) [inline resources/jscomposition/cases/casesDetail/components/TimelineCases.vue:169]: remove blank line

File: resources/jscomposition/cases/casesMain/components/.prettierrc
Comment (fagubla) [inline resources/jscomposition/cases/casesMain/components/.prettierrc:1]: is this file necessary?

File: upgrades/2024_10_08_113301_populate_comments_case_number.php [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+     * Returning void or null denotes the checks were successful.
+     *
+     * @return void
+     *
+     * @throws RuntimeException
+     */
+    public function preflightChecks()
+    {
+        //
+    }
```
Comment (fagubla) [inline upgrades/2024_10_08_113301_populate_comments_case_number.php:28]: this method is empty, is it ok?

**Review by fagubla — CHANGES_REQUESTED** [review]

File: upgrades/2024_10_08_113301_populate_comments_case_number.php (hunk 9 above)
Comment (pmPaulis) [reply upgrades/2024_10_08_113301_populate_comments_case_number.php:28]: This method was created from when we create a migrator

**Review by fagubla — APPROVED** [review]
